src/vm/node_modules/VM.js

```diff
@@ -2359,14 +2391,34 @@ function createVolume(volume, opts, callback)
         refreserv = 'auto';
     }
 
-    createResizeDeleteVolume({
-        log: log,
-        zpds: opts.zpds,
-        volname: volume.zfs_filesystem,
-        newsize: size * 1024 * 1024,
-        getSpaceDeltaFunc: cloning ? getCloneMdSize : getNewMdSize,
-        volChangeFunc: cloning ? cloneVolumeFunc :  createVolumeFunc
-    }, callback);
+    if (cloning) {
+        volblocksizeCb();
+    } else {
+        // we only (potentially) change the volblocksize for new volumes,
+        // since existing volumes (i.e. when cloning) already have a
+        // volblocksize
+        getOptimalVolblocksize(log, volblocksizeCb);
+    }
+
+    function volblocksizeCb(err, vbs) {
+        if (err) {
+            callback(err);
+            return;
+        }
+
+        if (vbs) {
+            volume.block_size = volume.block_size || vbs;
```

Comment:
This line is a bit unreadable to me.  Is this a (overly?) clever way of saying, "If you've volume.block_size set, and it's not set to zero, use it; otherwise use 'vbs'?"

Reply:
Yep. I'm used to the idiom, but if you'd prefer a more traditional formulation let me know,

Reply:
Yeah, this is pretty common in JS.